reflex/vars.py

```diff
@@ -1151,6 +1151,59 @@ def foreach(self, fn: Callable) -> Var:
             _var_is_local=self._var_is_local,
         )
 
+    def range(self, start: Var | int = 0) -> Var:
+        """Return an array of indicies from from start ot this var.
+
+        Args:
+            start: The start of the range.
+
+        Returns:
+            A var representing range operation.
+
+        Raises:
+            TypeError: If the var is not an int.
+        """
+        if self._var_type != int:
+            raise TypeError(f"Cannot get range on non-int var {self._var_full_name}.")
+        if isinstance(start, int):
+            start = Var.create_safe(start)
+        if start._var_type != int:
+            raise TypeError(
+                f"Cannot get start of range on non-int var {start._var_full_name}."
+            )
+
+        return BaseVar(
+            _var_name=f"Array({self._var_full_name}).fill(0).map((_, i) => i + {start})",
+            _var_type=list[self._var_type],
+            _var_is_local=False,
+        )
+
+    def forrange(self, fn: Callable) -> Var:
```

Comment:
I wonder if it makes more sense to introduce a new var if they do `rx.range(i)` which can match the API of the normal Python `range` operator. Then people can do something like

```python
rx.foreach(rx.range(5), my_function)
```

We can even then support start/stop indices

```python
rx.foreach(rx.range(1, 10), my_function)
```

Just because I think we should try to match the Python API as much as possible and not introduce new function names.